include/SSkinMeshBuffer.h

```diff
@@ -328,7 +328,22 @@ struct SSkinMeshBuffer : public IMeshBuffer
 	}
 
 	//! append the vertices and indices to the current buffer
-	void append(const void* const vertices, u32 numVertices, const u16* const indices, u32 numIndices) override {}
+	void append(const void* const vertices, u32 numVertices, const u16* const indices, u32 numIndices) override {
+		if (vertices == getVertices())
+			return;
```

Comment:
Is this correct? The docstring does not say that the vertices will not be appended if they happen to be equal to the current buffer.

Reply:
This implementation originates from another mesh buffer class; Irrlicht had left this unimplemented without so much as a note or a throws :unamused:

It doesn't make much sense to append the buffer to itself. It would probably be better to throw an error, because it's unclear what the proper behavior should be.

That said, I'm not a fan of these "append" methods. We should ideally probably get rid of them entirely, at least as far as gltf is concerned. This means adding a comment and a throwing body (because Irrlicht's misguided interfaces require us to provide this method :unamused:; refactoring that is probably out of scope for this PR, I believe).

If they were used as intended (to actually append), we might incur poor performance.

But we effectively only use this to set the contents once. It would be cleaner to let us a pass a mesh buffer to `addMeshBuffer`; the `SMeshBuffer` constructor should allow us to initialize the mesh buffer, rather than requiring us to modify an empty buffer.

Reply:
Change of plans, it's probably too much work to refactor this to be clean; I don't want to mess with Irrlicht that much yet. I'll just fix these issues and call it a day as far as gltf is concerned, Irrlicht cleanup can come later; I'd probably start that by eliminating dead / redundant code.

Reply:
Made it throw of a `std::logic_error`. Same for the not yet implemented second method, which used to just be empty before.

Reply:
Ah, I see. If it's done like this in the other class, then we should keep it that way for now.